src/librbd/ExclusiveLock.h

```diff
@@ -17,7 +17,7 @@ class ExclusiveLock : public ManagedLock<ImageCtxT> {
   }
 
   ExclusiveLock(ImageCtxT &image_ctx);
-
+  ~ExclusiveLock() = default;
```

Comment:
this is not necessary, right? as `LeaderLock` will have a virtual default dtor anyway. and so does its child classes if any. we can even mark `LeaderLock` `final` if we want to be more explicit.